llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2828,7 +2829,7 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpArrayLength>(SPIRVValue *
   const StructLayout *const structLayout = m_m->getDataLayout().getStructLayout(structType);
   const unsigned offset = static_cast<unsigned>(structLayout->getElementOffset(remappedMemberIndex));
   Value *const offsetVal = getBuilder()->getInt32(offset);
-  Value *const arrayBytes = getBuilder()->CreateGetBufferDescLength(pStruct, offsetVal);
+  Value *const arrayBytes = getBuilder()->create<BufferLengthOp>(pStruct, offsetVal);
```

Comment:
Further to comments on #2023, I see you are starting to use a dialect to replace and bypass the Builder interface. I misunderstood earlier discussion; I thought you were going to start with purely internal LGC stuff.

The question is: do we want to lose the "builder direct" path like this? The idea is that that would have saved a bit of compilation time, at the expense of not giving you semi-stable IR at the input-to-lgc level. But:

1. We have never used it.
2. I guess the dialect thing has potential to be less wasteful than the "builder recorder and replay" thing in that you don't need to lower everything in a replay phase as soon as you get in to LGC, only to need to lower some of it again later on.

So is everyone happy for this fundamental change to the lgc interface to start happening here?

Reply:
I removed "builder direct" mode a while ago.